src/actions/onboardingActions.js

```diff
@@ -172,9 +175,15 @@ const finishRegistration = async ({
     // create smart wallet account only for new wallets
     const createNewAccount = !isImported;
     await dispatch(initSmartWalletSdkAction(privateKey));
-    await dispatch(importSmartWalletAccountsAction(privateKey, createNewAccount));
+    await dispatch(importSmartWalletAccountsAction(privateKey, createNewAccount, initialAssets));
   }
 
+  const { accounts: { data: accounts } } = getState();
+
+  accounts.forEach(async (acc) => {
```

Comment:
This won't work :( you need to use 
`await Promise.all(accounts.map(acc => dispatch(restoreTransactionHistoryAction(acc.id, userInfo.walletId))))`

Reply:
Ahh, true, thanks!